modules/internal/localeModels/knl/LocaleModel.chpl

```diff
@@ -319,7 +319,6 @@ module LocaleModel {
 
     iter getChildren() : locale {
       halt("No children to iterate over.");
-      yield nil;
```

Comment:
This is also surprising, as we used to disallow iterators without any `yield` statements at all. OK to leave this way since it passes testing.